src/main/java/uk/gov/hmcts/darts/common/repository/HearingRepository.java

```diff
@@ -31,6 +31,14 @@ public interface HearingRepository extends JpaRepository<HearingEntity, Integer>
     )
     List<HearingEntity> findByCaseIds(List<Integer> caseIds);
 
+    @Query("""
+        SELECT h.id FROM HearingEntity h
+        JOIN h.eventList event
+        WHERE event.id = :eventId
+        """
+    )
+    List<Integer> findHearingIdsByEventId(Integer eventId);
```

Comment:
Are you going to create a test for this query like EventRepositoryTest?

Reply:
done